Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift

```diff
@@ -30,72 +30,73 @@ extension VideoPlayer {
         @StateObject
         private var overlayTimer: TimerProxy = .init()
 
-        var body: some View {
-            ZStack {
-
-                MainOverlay()
-                    .visible(currentOverlayType == .main)
-
+        @ViewBuilder
+        private var currentOverlay: some View {
+            switch currentOverlayType {
+            case .chapters:
+                ChapterOverlay()
+            case .confirmClose:
                 ConfirmCloseOverlay()
-                    .visible(currentOverlayType == .confirmClose)
-
+            case .main:
+                MainOverlay()
+            case .smallMenu:
```

Comment:
Also, so that this is attributable in the future, this was changed from a `ZStack` because

1. We only show one overlay at a time
2. With multiple overlays mounted, each of them fight for focus. Specifically, having the `MainOverlay` mounted with the `SmallMenuOverlay` caused focus conflicts that I couldn't resolve.